app/controllers/active_storage/blobs/proxy_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+# Proxy files through application with authentication.
+# This avoids making the S3 bucket where files are stored public
+
+# Overwriting
+# https://github.com/rails/rails/blob/6-1-stable/activestorage/app/controllers/active_storage/blobs/proxy_controller.rb
+class ActiveStorage::Blobs::ProxyController < ActiveStorage::BaseController
+  include ActiveStorage::SetBlob
+  include ActiveStorage::SetHeaders
+
+  def show
+    redirect_to('/') && return unless current_user&.id == @blob.attachments.first.record.user_id
```

Comment:
Are there situations where @blob might not be what we expect? Is it worth using `dig` here, and if fail also redirect?

Reply:
I've added safe navigation chain to `@blob.attachments.first...` to guard against a blob without an attachment.

Not sure what you mean by using `dig` since `@blob` is an AR object/model the calls aren't hash lookups they are associations.